### PR TITLE
Open encrypted file and write back the category to the transaction

### PR DESCRIPTION
**Pull request by MotafokaBR**
- Included an optional environment variable to get the password for encrypted files

- Changed the download step to open encrypted files
- Included an extra step to write back the transaction with the ChatGPT category

### CODE REVIEW

**Comment by MotafokaBR** [comment]
I have been using weekly for the past few weeks and it works wonders.

File: suggest-cats.js [hunk 1]
```diff
@@ -82,10 +88,17 @@ const categorize = (transaction) => {
...
+        messages.push(categorize(res.data[i].imported_payee))
+        const response = await chat(messages);
+        console.log(`response: ${response.choices[0].message.content}`);
+        
+        let newCategory = existing_categories.find(cat => cat.name === response.choices[0].message.content);
+        if (newCategory) {
+          await api.updateTransaction(res.data[i].id, { category: newCategory.id });
+        }
+        messages.pop();
+      }
```
Comment (psybers) [inline suggest-cats.js]: Can you remove this from this PR?  I planned to allow updates, but I wanted to make those optional and have the possibility of being interactive (to confirm each one before updating).

**Review by psybers — CHANGES_REQUESTED** [review]

**Review by MotafokaBR — COMMENTED** [review]
Not sure if I'm doing this right

**Comment by psybers** [comment]
Looks good, thank you!

**Pull request merged**